source/nsdynmemLIB/nsdynmemLIB.c

```diff
@@ -266,19 +214,28 @@ void *ns_dyn_mem_alloc(int16_t alloc_size)
 
 
         if (mem_stat_info_ptr) {
-            alloc_size = (alloc_sector * sizeof(int));
+            // If alloc fail needs to collect size of failed alloc,
+            // this code needs to be revisited
+            alloc_size = ((alloc_sector + 2) * sizeof(int));
+
             if (retval) {
                 //Update Allocate OK
                 dev_stat_update(DEV_HEAP_ALLOC_OK, alloc_size);
 
             } else {
                 //Update Allocate Fail
-                dev_stat_update(DEV_HEAP_ALLOC_FAIL, alloc_size);
+                dev_stat_update(DEV_HEAP_ALLOC_FAIL, 0);
             }
         }
         platform_exit_critical();
     }
     return retval;
+}
+
+void *ns_dyn_mem_alloc(int16_t alloc_size)
+{
+#ifndef STANDARD_MALLOC
```

Comment:
Could you not have a single STANDARD_MALLOC ifdef inside the internal function? Save having two copies of the "call malloc" code?